sky/sdk/lib/widgets/block_viewport.dart

```diff
@@ -155,8 +198,43 @@ class BlockViewport extends RenderObjectWrapper {
     }
   }
 
+  // Build the widget at index, and use its maxIntrinsicHeight to fix up
+  // the offsets from index+1 to endIndex. Return the newWidget.
+  Widget _recomputeOffsets(int index, int endIndex, BoxConstraints innerConstraints) {
+    LayoutCallbackBuilderHandle handle = enterLayoutCallbackBuilder();
```

Comment:
restructure the code so that enterLayoutCallbackBuilder() only occurs in one place.